accepted/0003-interactive-audit-resolver.md

```diff
@@ -0,0 +1,155 @@
+- `.npmrc` is not a good option because it doesn't live in the repository and being a dotfile is easy to miss, so a developer could overlook it affecting the audit. Also using it for the purpose of storing resolutions to project specific issues seems counter-intuitive.
+
+A separate file referred to as `audit-resolve.json` has the benefit of being single purpose, easy to track and audit, easy to version and migrate between versions of `npm` and comfortable for the users to review in git history and pull requests.
+
+**postpone, remove**
+Other options are helpers for more elaborate actions a developer would take when confronted with an issue they can't or don't want to fix. 
+
+Why is postpone useful at all? It's designed to build a secure development culture where one didn't yet form. Without it, a developer under time pressure would mark an issue as ignored with intention to un-mark it later. 
+While shipping with a known vulnerability is a bad practice, NPM's mission with the community should be to empower people to build more secure products and trust their skill and understanding of their project's particular needs. We should also aspire to help teams introduce more secure workflows effortlessly, so letting a build pass without risking compromising security long-term is a win. 
+
+*audit-resolver* could perform more actions, like let the user remove a package entirely or help find a patch in the future.
+
+### Alternatives
+
+Currently the only alternative is to manage the issues manually and keep track of the resolutions using conventions created by a team individually. Ignoring dev dependencies and certain level of severity will be provided for `npm audit` so it lowers the barrier of entry to using it in a CI system. 
+
+Paid alternatives for managing security of a node.js project are available, including Snyk, with focus on providing patches to their customers.
+
+## Implementation
+
+prototype of a working *audit-resolver* (in production use) https://www.npmjs.com/package/npm-audit-resolver
+
+Core capability covering reading, parsing, validating and representing the `audit-resolve.json` file was extracted from npm-audit-resolver into https://www.npmjs.com/package/audit-resolve-core - API of that package to be discussed. Functionality is there.
+
+The implementation is, and should remain, runnable standalone as a separate package with minor wrapping code - useful for testing new features without bundling unfinished work with npm cli versions and therefore node.js
+
+
+### audit-resolve.json
+
+Audit resolution file format:
+map key-value where key is the advisory number concatenated with package path
+
+```js
+{
+  "version": 1,
+  "decisions": {
+    "ADVISORY_NUMBER|DEPENDENCY_PATH":{
+      "decision": "RESOLUTION_TYPE",
+      "reason": "Reason provided by the person making the decision (optional)",
+      "madeAt": timestamp
+      "expiresAt": timestamp
```

Comment:
Can this please also (only?) support a human readable ISO date string?

Use case: extending the expiry time without using any tooling (an extremely common operation, from experience with `.snyk` policy files...)

Reply:
Personally I believe it should not be limited to ISO string because of the ending I always mess up. 
2020-09-21 should also work if we want to cater to humans :)


Next iteration of the core package that reads and writes the resolve file will have that.

Thanks for the data point!